app/models/miq_cockpit_ws_worker/runner.rb

```diff
@@ -0,0 +1,65 @@
+class MiqCockpitWsWorker::Runner < MiqWorker::Runner
+  BINDING_ADDRESS = ENV['BINDING_ADDRESS'] || (Rails.env.production? ? nil : "0.0.0.0")
+
+  def self.start_worker(*args)
+    runner = MiqCockpitWsWorker::Runner.new(*args)
+
+    runner.prepare
+    runner.start_cockpit_ws
+    runner.run
+  end
+
+  def do_work
+  end
+
+  def pid_file
+    Rails.root.join("tmp/pids/cockpit-ws.pid")
```

Comment:
The pidfile class is a little different, because the pid we are writing isn't our process pid. Honestly I'd rather do something like 

```
runner.prepare
fork do
  prctl(PR_SET_PDEATHSIG, SIGTERM)
  runner.run
end
runner.start_cockpit_ws
```

That way there isn't all this extra pid handling, but i wasn't able to find a way to do PR_SET_PDEATHSIG in ruby.